daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/NodeInfo.scala

```diff
@@ -452,9 +454,30 @@ object NodeInfo extends Enum {
       }
     }
 
+    trait PrimThrowsIllegalArgument {
```

Comment:
I'm not sure this separatation between IllegalArgument and NumberFormatException gains a whole lot. It does mean we are a little more strict on what we expect to be thrown, but at the cost of duplicating code and I think making it a little harder to read. I don't think we lose much by catching both execptions in a single fromXMLString call.

Reply:
So I'm currently planning on leaving this in as I am now having PrimNumeric fromNumber/XMLString perform range checks too.